docs/cg.rst

```diff
@@ -0,0 +1,81 @@
+Managing the computation graph
+==============================
+
+Theano constructs computation graphs of mathematical expressions. Bricks help
+you :doc:`build these graphs <bricks_overview>`, but they do more than that.
+When you apply a brick to a Theano variable, it automatically *annotates* this
+Theano variable, in two ways:
+
+* It defines the *role* this variable plays in the computation graph e.g. it will
+  label weights matrics and biases as parameters, keep track of which variables
+  where the in- and outputs of your bricks, and more.
+* It constructs *auxiliary variables*. These are variables which are not an
+  output of your brick, but might still be of interest. For example, if you are
+  training a neural network, you might be interested to now the norm of your
+  weight matrices, so Blocks attaches these as auxiliary variables to the graph.
```

Comment:
That sounds like a good idea, it would be a nice by-product, although it actually wasn't the main goal of this endeavor. I mostly wanted to get closer to the idea of a "managed Theano graph" without the need for using bricks.